src/coreComponents/linearAlgebra/interfaces/petsc/PetscSuperlu.hpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file PetscSuperlu.hpp
+ */
+
+#ifndef GEOSX_LINEARALGEBRA_INTERFACES_PETSCSUPERLU_HPP_
+#define GEOSX_LINEARALGEBRA_INTERFACES_PETSCSUPERLU_HPP_
+
+#include "common/DataTypes.hpp"
+#include "linearAlgebra/interfaces/petsc/PetscMatrix.hpp"
+#include "linearAlgebra/utilities/LinearSolverParameters.hpp"
+
+namespace geosx
+{
+
+/**
+ * @brief Creates the SuperLU_Dist data structure
+ * @param[in] matrix the PetscMatrix object
+ * @param[in] params the linear solver parameters
+ */
+void SuperLU_DistSetFromOptions( PetscMatrix const & matrix,
+                                 LinearSolverParameters const & params );
+
+/**
+ * @brief Converts from GEOSX to SuperLU_Dist columns permutation option
+ * @param[in] value the GEOSX option
+ * @return the SuperLU_Dist option
+ */
+string const & PetscGetColPermType( string const & value );
+
+/**
+ * @brief Converts from GEOSX to SuperLU_Dist rows permutation option
+ * @param[in] value the GEOSX option
+ * @return the SuperLU_Dist option
+ */
+string const & PetscGetRowPermType( string const & value );
```

Comment:
Same comment as in hypre, let's not expose these unless needed.

Reply:
You're right.